lib/service/awsoidc.go

```diff
@@ -70,18 +69,13 @@ func (process *TeleportProcess) initAWSOIDCDeployServiceUpdater() error {
 		return nil
 	}
 
-	// If criticalEndpoint or versionEndpoint are empty, the default stable/cloud endpoint will be used
-	var criticalEndpoint string
-	var versionEndpoint string
-	if automaticupgrades.GetChannel() != "" {
-		criticalEndpoint, err = url.JoinPath(automaticupgrades.GetChannel(), "critical")
-		if err != nil {
-			return trace.Wrap(err)
-		}
-		versionEndpoint, err = url.JoinPath(automaticupgrades.GetChannel(), "version")
-		if err != nil {
-			return trace.Wrap(err)
-		}
+	// TODO: use the proxy channel if available?
+	// This would require to pass the proxy configuration there, but would avoid
+	// future inconsistencies: if the proxy is manually configured to serve a
+	// static version, it will not be picked up by the AWS OIDC deploy updater.
+	upgradeChannel, err := automaticupgrades.NewDefaultChannel()
```

Comment:
I want to do this, but we have no guarantee the `Channels` is already initialized as it needs the server features to do so.

Reply:
Ah, then can we check first if the channels are initialized, and use `NewDefaultChannel` if they are not initialized? Also this process only runs if `AutomaticUpgrades` feature is enabled.

Reply:
THis would cause a race and a non-deterministic behaviour. The first time you might get the default channel, while the next call you get the one from the config. Also, the Channel slice is not protected by a mutex and is not thread safe, the writes operations should happen strictly before we start to read.